src/ott/problems/linear/potentials.py

```diff
@@ -258,3 +264,125 @@ def epsilon(self) -> float:
 
   def tree_flatten(self) -> Tuple[Sequence[Any], Dict[str, Any]]:
     return [self._f, self._g, self._prob, self._f_xx, self._g_yy], {}
+
+
+def plot_ot_map(
+    learned_potentials: DualPotentials,
+    source: jnp.ndarray,
+    target: jnp.ndarray,
+    inverse: bool = False,
+    ax: Optional[matplotlib.axes.Axes] = None,
+    **kwargs: Any,
+) -> Tuple[matplotlib.figure.Figure, matplotlib.axes.Axes]:
+  """Plot data and learned optimal transport map.
+
+  Args:
+    learned_potentials: the potentials object for the map
+    source: samples from the source measure
+    target: samples from the target measure
+    inverse: if the inverse map from the potentials
+      should be used
+    ax: axis to add the plot to
+
+  Returns: matplotlib figure and axis with the plots
+  """
+
+  def draw_arrows(a: np.ndarray, b: np.ndarray) -> None:
+    ax.arrow(
+        a[0], a[1], b[0] - a[0], b[1] - a[1], color=[0.5, 0.5, 1], alpha=0.3
+    )
+
+  grad_state_s = learned_potentials.transport(source, forward=not inverse)
+
+  if ax is None:
+    fig = plt.figure(facecolor="white")
+    ax = fig.add_subplot(111)
+  else:
+    fig = ax.get_figure()
+
+  if not inverse:
+    label_src, label_tgt = 'source', 'target'
+    label_transport = r"$\nabla f(source)$"
+  else:
+    label_src, label_tgt = 'target', 'source'
+    label_transport = r"$\nabla g(source)$"
+
+  ax.scatter(
+      source[:, 0],
+      source[:, 1],
+      color="#1A254B",
+      alpha=0.5,
+      label=label_src,
+      **kwargs,
+  )
+  ax.scatter(
+      target[:, 0],
+      target[:, 1],
+      color="#A7BED3",
+      alpha=0.5,
+      label=label_tgt,
+      **kwargs,
+  )
+  ax.scatter(
+      grad_state_s[:, 0],
+      grad_state_s[:, 1],
+      color="#F2545B",
+      alpha=0.5,
+      label=label_transport,
+      **kwargs,
+  )
+
+  fig.legend(ncol=3, loc="upper center")
+
+  for i in range(source.shape[0]):
+    draw_arrows(source[i, :], grad_state_s[i, :])
+
+  return fig, ax
```

Comment:
Consider returning just the `ax`, since user can always do `ax.get_figure()`.

Reply:
It is a nice convenience to just do this in the notebooks/everywhere else it's called instead of manually calling `ax.get_figure()` everywhere:

![image](https://user-images.githubusercontent.com/707462/217888852-fd1ac61a-4a9e-4b1a-9d3a-62a28f3ff44e.png)

Reply:
Ok, let's return both fig and ax.